core/src/test/java/org/bitcoinj/core/BitcoinSerializerTest.java

```diff
@@ -163,10 +163,12 @@ private void testCachedParsing(boolean lazy)  throws Exception {
     public void testHeaders1() throws Exception {
         BitcoinSerializer bs = new BitcoinSerializer(MainNetParams.get());
 
-        HeadersMessage hm = (HeadersMessage) bs.deserialize(ByteBuffer.wrap(HEX.decode("f9beb4d9686561" +
+        String headersMessageBytesHex = "f9beb4d9686561" +
                 "646572730000000000520000005d4fab8101010000006fe28c0ab6f1b372c1a6a246ae6" +
                 "3f74f931e8365e15a089c68d6190000000000982051fd1e4ba744bbbe680e1fee14677b" +
-                "a1a3c3540bf7b1cdb606e857233e0e61bc6649ffff001d01e3629900")));
+                "a1a3c3540bf7b1cdb606e857233e0e61bc6649ffff001d01e3629900";
+        byte[] headersMessageBytes = HEX.decode(headersMessageBytesHex);
+        HeadersMessage hm = (HeadersMessage) bs.deserialize(ByteBuffer.wrap(HEX.decode(headersMessageBytesHex)));
```

Comment:
Shouldn't you wrap headersMessageBytes here? Seems like there's a common subexpression.